eloquentjs/4_3_list.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * Returns list from array
+ * @param {*} inputArray 
+ * @returns 
+ */
+function arrayToList(inputArray) { 
+    if (!Array.isArray(inputArray)) { 
+        throw "input must be array";
+    }
+
+    let currentIndex = inputArray.length-1;
+
+    function innerArrayToList(inputArray, list = null) {
```

Comment:
`inputArray` is the same as the `inputArray` this inner function closes over, so you can drop it from the function signature. I'd move currentIndex into the function arguments, though.